Source/Scene/IonImageryProvider.js

```diff
@@ -156,10 +155,23 @@ define([
         this._errorEvent = new Event();
 
         var that = this;
-        this._readyPromise = endpointResource.fetchJson()
+        var endpointResource = IonResource._createEndpointResource(assetId, options);
+
+        // A simple cache to avoid making repeated requests to ion for endpoints we've
+        // already retrieved. This exists mainly to support Bing caching to reduce
+        // world imagery sessions, but provides a small boost of performance in general
+        // if constantly reloading assets
+        var cacheKey = JSON.stringify(options);
```

Comment:
Your cacheKey will come out differently when options are specified in a different order.  Probably not important but that's why in most existing places where we make cache keys we stringify an array instead.

Reply:
Good catch. I agree it's an edge case, but since I found a one-liner to take care of it:

```js
var cacheKey = JSON.stringify(options, Object.keys(options).sort());
```

Reply:
Actually, now that I think about it, a direct array would be better since I know what the 3 properties are ahead of time.